src/Broadway/Snapshotting/Snapshotable.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/**
+ * This file is part of the broadway/broadway package.
+ *
+ *  (c) Qandidate.com <opensource@qandidate.com>
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ */
+
+namespace Broadway\Snapshotting;
+
+interface Snapshotable
```

Comment:
Disclaimer: I'm not a Broadway user, so this comment only adds my 2c

Do you really need such an interface? It requires extra effort to implement and test this functionality. It adds a public method to a domain aggregate that should not be aware of snapshots because it is an infrastructure concern.

Also this introduces a second way of recreating internal state of the aggregate. Where you have only "event apply" methods before you now have an additional "apply snapshot" mechanism. This could lead to strange bugs if the aggregate behaves different with or without snapshot applied.

Why not simply serialize an aggregate using PHP build in functionality? This would be a real snapshot, not a user generated one.